mscore/plugin/pluginUpdater.cpp

```diff
@@ -0,0 +1,866 @@
+                              continue;
+                        if (isCompatibleRelease(release_obj.value("target_commitish").toString())) {
+                              // get "id" and "zipball_url"
+                              release_id = release_obj.value("id").toInt();
+                              link = release_obj.value("zipball_url").toString();
+                              return true;
+                              }
+                        else
+                              continue;
+                        }
+                  else
+                        continue;
+                  }
+            return false;
+            }
+      else
+            return false;
+      }
+
+static inline bool isNotFoundMessage(const QJsonObject& json_obj) {
+      return json_obj["message"].toString() == "Not Found";
+      }
+static inline QString filenameBaseFromPageURL(QString page_url) {
+      Q_ASSERT(page_url.contains("/project/"));
+      if (page_url.contains("/en/project/"))
+            return page_url.remove("/en/project/");
+      return page_url.remove("/project/");
+      }
+static inline QString getExtFromURL(QString& direct_link) {
+      QString possible_ext = direct_link.split(".").last();
+      // We can assume that most extensions contain only 
+      // alphanumeric and plus "_", and other connector punctuation chars
+      if (possible_ext.contains(QRegularExpression("\\W")))
+            return {};
+      return possible_ext;
+      }
+
+//---------------------------------------------------------
+//   parsePluginRepo
+//---------------------------------------------------------
+void ResourceManager::parsePluginRepo(QByteArray html_raw)
+      {
+      QByteArray table_start("<table");
+      QByteArray table_end("table>");
+      int table_start_idx = html_raw.indexOf(table_start);
+      int table_end_idx = html_raw.indexOf(table_end) + 6;
+      QByteArray table_raw = html_raw.mid(table_start_idx, table_end_idx - table_start_idx);
+      QDomDocument table_xml;
+      if (!table_xml.setContent(table_raw))
+            qDebug("Fail to parse the HTML table from the repo page.");
+      QDomElement body = table_xml.elementsByTagName("tbody").item(0).toElement();
+      QDomNodeList tr_list = body.elementsByTagName("tr");
+      pluginsTable->clearSpans();
+      pluginsTable->setRowCount(tr_list.length());
+      pluginsTable->setItem(0, 0, nullptr);
+      QStringList all_categories;
+      for (int row = 0; row < tr_list.length(); row++) {
+            int col = 0;
+            QDomElement tr_node = tr_list.item(row).toElement();
+            QDomNodeList td_list = tr_node.elementsByTagName("td");
+            Q_ASSERT(td_list.length() == 4);
+            QString page_url = td_list.item(0).toElement().firstChildElement("a").attribute("href");
+            QString name = td_list.item(0).toElement().text();
+            std::tuple<bool,bool,bool> compat = compatFromString(td_list.item(1).toElement().text());
```

Comment:
Compiler warning: warning: variable `compat` set but not used [-Wunused-but-set-variable]

Reply:
Indeed. Before the info was displayed in the table. Now it's not used anymore